r/R/dplyr.R

```diff
@@ -480,6 +480,18 @@ build_function_list <- function(FUN) {
     between = function(x, left, right) {
       x >= left & x <= right
     },
+    sd = function(x, na.rm = FALSE){
+      if (!na.rm && x$null_count > 0) {
+        return(Scalar$create(NA_real_))
+      }
```

Comment:
I'm not sure if this is wrong somehow, as when I call the code below, I get the warning `Warning: Expression mass - sd(mass, na.rm = FALSE) not supported in Arrow; pulling data into R`, but yet when I set `na.rm` to TRUE, I don't.
```
Table$create(starwars) %>%
     select(name, mass, species) %>%
     mutate(mass_diff = mass-sd(mass, na.rm = FALSE)) %>%
     collect()
```

Reply:
We don't support aggregations in our dplyr backend yet, so this should never succeed. If `sd()` doesn't cleanly and always error when called on an arrow Expression, we should force it to--see the "fail" handling inside of `arrow_eval` where this is done for `mean`.

Reply:
Done.